### PR TITLE
Setltest heaters

### PR DESCRIPTION
**Pull request by RadekVana**
BFW-1516, BFW-1579, BFW-1578

### CODE REVIEW

File: src/common/selftest/selftest_MINI.cpp [hunk 1]
```diff
@@ -387,7 +387,7 @@ CSelftestPart::CSelftestPart()
 CSelftestPart::~CSelftestPart() {
 }
 
-float CSelftestPart::GetProgress() const {
+float CSelftestPart::GetProgress() {
```
Comment (DRracer) [inline src/common/selftest/selftest_MINI.cpp:390]: why isn't this const anymore?

File: src/common/selftest/selftest_heater.cpp [hunk 2]
```diff
@@ -62,33 +160,36 @@ bool CSelftestPart_Heater::Loop() {
...
+        break;*/
     }
     case spsMeasure:
-        if ((Selftest.m_Time - m_Time) < TEMP_MEASURE_CYCLE_DELAY) {
+        if (int(m_EndTime - Selftest.m_Time) > 0) {
+            m_Temp = getTemp();
+            return true;
+        }
+
+        /* if ((Selftest.m_Time - m_Time) < TEMP_MEASURE_CYCLE_DELAY) {
```
Comment (DRracer) [inline src/common/selftest/selftest_heater.cpp:176]: what about this commented code? Is it still relevant?

File: src/common/selftest/selftest_MINI.cpp (hunk 1 above)
Comment (RadekVana) [reply src/common/selftest/selftest_MINI.cpp:390]: because child changes interval variable

File: src/common/selftest/selftest_heater.cpp (hunk 2 above)
Comment (RadekVana) [reply src/common/selftest/selftest_heater.cpp:176]: it is, did not want to remove Roberts code

**Review by XPila — APPROVED** [review]

**Pull request merged**